src/swarmClient/ecdsa_p256.js

```diff
@@ -101,18 +96,20 @@ const import_public_key_from_base64 = pub_key_base64 => {
     // - 1.3.132.0.10 secp256k1 (SECG (Certicom) named elliptic curve)
     // - The length and type of the remaining bitstring.
 
-    // To derive this, take a sample secp256k1 public key from OpenSSL and run it through
+    // To derive this, take a sample ec public key from OpenSSL and run it through
     // an ASN.1 decoder such as https://lapo.it/asn1js/.
 
-    const header = key_hex.substring(0, 46);
 
-    assert.equal(header, "3056301006072a8648ce3d020106052b8104000a034200",
-        "ECDSA Signature Verification: public key header is malformed for secp256k1. This is the public key you're trying to decode: \"" + pub_key_base64 + '"');
+    const header = key_hex.substring(0, 52);
+
+    assert.equal(header, "3059301306072a8648ce3d020106082a8648ce3d030107034200",
```

Comment:
Out of curiosity, is there not some `verify` method or something in the libraries we're using? Why are we doing this ourselves? (question applies to other places where we're using magic bytes and substrings)

Reply:
I have found no javascript library that can generically interpret private/public keys encoded in PKCS #1/6 (Public Key Cryptography Standards). The library I'm using here, elliptic, imports encodings of the points, which is a subset of the whole key. The rest describes things like the algorithm being used, the curve params, etc.